controllers/lustrefilesystem_controller.go

```diff
@@ -197,52 +281,55 @@ func createOrUpdatePersistentVolume(ctx context.Context, r *LustreFileSystemReco
 			},
 		}
 
-		volumeMode := corev1.PersistentVolumeFilesystem
-		pv.Spec.VolumeMode = &volumeMode
-
 		return nil
 	}
 
-	return ctrl.CreateOrUpdate(ctx, r.Client, pv, mutateFn)
-}
+	result, err := ctrl.CreateOrUpdate(ctx, r.Client, pv, mutateFn)
+	if err != nil {
+		return nil, err
+	}
+
+	if result != controllerutil.OperationResultNone {
+		log.FromContext(ctx).Info("PersistentVolume", "object", client.ObjectKeyFromObject(pv).String(), "result", result)
+	}
 
-func createOrUpdatePersistentVolumeClaim(ctx context.Context, r *LustreFileSystemReconciler, fs *v1alpha1.LustreFileSystem, ownerRef *metav1.OwnerReference, log logr.Logger) (controllerutil.OperationResult, error) {
-	log = log.WithValues("PersistentVolumeClaim", fs.Name+PersistentVolumeClaimSuffix)
+	return pv, nil
+}
 
+func (r *LustreFileSystemReconciler) deleteAccess(ctx context.Context, fs *v1alpha1.LustreFileSystem, namespace string, mode corev1.PersistentVolumeAccessMode) error {
 	pvc := &corev1.PersistentVolumeClaim{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      fs.Name + PersistentVolumeClaimSuffix,
-			Namespace: fs.Namespace,
+			Name:      fs.PersistentVolumeClaimName(namespace, mode),
+			Namespace: namespace,
 		},
 	}
 
-	mutateFn := func() error {
-		pvc.ObjectMeta.SetOwnerReferences([]metav1.OwnerReference{*ownerRef})
-
-		pvc.Spec.StorageClassName = &fs.Spec.StorageClassName
-		// Reserve this PVC for the matching PV.
-		pvc.Spec.VolumeName = fs.Name + PersistentVolumeSuffix
-
-		pvc.Spec.AccessModes = []corev1.PersistentVolumeAccessMode{
-			corev1.ReadWriteMany,
+	log.FromContext(ctx).Info("Deleting PersistentVolume", "object", client.ObjectKeyFromObject(pvc).String())
```

Comment:
The log message says it's a persistent volume, but you're deleting a PVC.
And in the log below you have these reversed.